webui/react/src/pages/ProjectDetails.tsx

```diff
@@ -162,42 +164,58 @@ const ProjectDetails: React.FC = () => {
       const states = (settings.state || []).map((state) => (
         encodeExperimentState(state as RunState)
       ));
-      const response = await getExperiments(
+      const experimentFilter: GetExperimentsParams = {
+        archived: settings.archived ? undefined : false,
+        labels: settings.label,
+        name: settings.search,
+        orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+        projectId: id,
+        sortBy: validateDetApiEnum(V1GetExperimentsRequestSortBy, settings.sortKey),
+        states: validateDetApiEnumList(Determinedexperimentv1State, states),
+        users: settings.user,
+      };
+      const pinnedExperimentsRequest = getExperiments(
         {
-          archived: settings.archived ? undefined : false,
-          labels: settings.label,
-          limit: settings.tableLimit,
-          name: settings.search,
-          offset: settings.tableOffset,
-          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
-          projectId: id,
-          sortBy: validateDetApiEnum(V1GetExperimentsRequestSortBy, settings.sortKey),
-          states: validateDetApiEnumList(Determinedexperimentv1State, states),
-          users: settings.user,
+          ...experimentFilter,
+          experimentFilter: { experimentIds: settings.pinned, restriction: 'RESTRICTION_INCLUDE' },
+          limit: 0,
+          offset: 0,
         },
         { signal: canceler.signal },
       );
-      setTotal(response.pagination.total ?? 0);
-      setExperiments((prev) => {
-        if (isEqual(prev, response.experiments)) return prev;
-        return response.experiments;
-      });
+      const otherExperimentsRequest = getExperiments(
+        {
+          ...experimentFilter,
+          experimentFilter: { experimentIds: settings.pinned, restriction: 'RESTRICTION_EXCLUDE' },
+          limit: settings.tableLimit - settings.pinned.length,
+          offset: settings.tableOffset -
+            (settings.tableOffset / settings.tableLimit) * settings.pinned.length,
+        },
+        { signal: canceler.signal },
+      );
+      const allRequests = [ pinnedExperimentsRequest, otherExperimentsRequest ];
+      const [ pinnedExpResponse, otherExpResponse ] = await Promise.all(allRequests);
+      setTotal(otherExpResponse.pagination.total ?? 0);
```

Comment:
fix: pagination gets messed up if you have too many experiments pinned compared to total experiments. On my local cluster I have 15 experiments and pinned 11 of them. I'm only able to see the 10 of the pinned experiments and I have no pagination controls or way to see any other experiments

Reply:
I thought I fixed the issue. Let me take a look

Reply:
I added limit for the number of pin items which is 5.